public/app/features/alerting/unified/components/contact-points/useContactPoints.tsx

```diff
@@ -7,48 +7,81 @@ import { produce } from 'immer';
  * 1. the alertmanager config endpoint where the definition of the receivers are
  * 2. (if available) the alertmanager receiver status endpoint, currently Grafana Managed only
+ * 3. (if available) additional metadata about Grafana Managed contact points
+ * 4. (if available) the OnCall plugin metadata
  */
-export function useContactPointsWithStatus(selectedAlertmanager: string) {
-  const isGrafanaManagedAlertmanager = selectedAlertmanager === GRAFANA_RULES_SOURCE_NAME;
+export function useContactPointsWithStatus() {
+  const { selectedAlertmanager, isGrafanaAlertmanager } = useAlertmanager();
+  const { installed: onCallPluginInstalled = false, loading: onCallPluginStatusLoading } = usePluginBridge(
+    SupportedPlugin.OnCall
+  );
 
   // fetch receiver status if we're dealing with a Grafana Managed Alertmanager
   const fetchContactPointsStatus = alertmanagerApi.endpoints.getContactPointsStatus.useQuery(undefined, {
-    // TODO these don't seem to work since we've not called setupListeners()
     refetchOnFocus: true,
     refetchOnReconnect: true,
     // re-fetch status every so often for up-to-date information
     pollingInterval: RECEIVER_STATUS_POLLING_INTERVAL,
     // skip fetching receiver statuses if not Grafana AM
-    skip: !isGrafanaManagedAlertmanager,
+    skip: !isGrafanaAlertmanager,
+  });
+
+  // fetch notifier metadata from the Grafana API if we're using a Grafana AM – this will be used to add additional
+  // metadata and canonical names to the receiver
+  const fetchReceiverMetadata = alertmanagerApi.endpoints.grafanaNotifiers.useQuery(undefined, {
+    skip: !isGrafanaAlertmanager,
   });
 
+  // if the OnCall plugin is installed, fetch its list of integrations so we can match those to the Grafana Managed contact points
+  const { data: onCallIntegrations, isLoading: onCallPluginIntegrationsLoading } =
+    onCallApi.endpoints.grafanaOnCallIntegrations.useQuery(undefined, {
+      skip: !onCallPluginInstalled || !isGrafanaAlertmanager,
+    });
+
   // fetch the latest config from the Alertmanager
   const fetchAlertmanagerConfiguration = alertmanagerApi.endpoints.getAlertmanagerConfiguration.useQuery(
-    selectedAlertmanager,
+    selectedAlertmanager!,
     {
       refetchOnFocus: true,
       refetchOnReconnect: true,
       selectFromResult: (result) => ({
         ...result,
-        contactPoints: result.data ? enhanceContactPointsWithStatus(result.data, fetchContactPointsStatus.data) : [],
+        contactPoints: result.data
+          ? enhanceContactPointsWithMetadata(
+              result.data,
+              fetchContactPointsStatus.data,
+              fetchReceiverMetadata.data,
+              onCallPluginInstalled ? onCallIntegrations ?? [] : null
+            )
+          : [],
       }),
     }
   );
 
-  // TODO kinda yucky to combine hooks like this, better alternative?
+  // we will fail silently for fetching OnCall plugin status and integrations
```

Comment:
What does it mean? What happens when we cannot get OnCall status?

Reply:
So if we fail to contact the onCall plugin (because it's either not installed, disabled or a network error happened) we still render the contact points list instead of showing an error on the page. The OnCall receiver will show up as "integration not found in OnCall"